editor/store/effects.js

```diff
@@ -306,6 +308,8 @@ export default {
 				return block;
 			} );
 			effects.push( resetBlocks( blocks ) );
+		} else if ( getDefaultBlockForPostFormat( post.format ) ) {
+			effects.push( insertBlock( createBlock( getDefaultBlockForPostFormat( post.format ) ) ) );
```

Comment:
I think it should be fine (and master is currently broken anyways, see #4836), but we should confirm that this doesn't cause the post to become flagged as dirty for a new post.

Should be fine because we're resetting dirtiness on `RESET_POST`, which is an effect occurring after this one.

Reply:
Unfortunately, the dirty checking mechanism is not working on this PR, so when a non-standard default post format is set the post is dirty from start. I don't think it's a problem specific to this PR, e.g: the same problem happens for custom post type templates. It looks like the reset post is not working as expected but more debugging will be required.